browser/playlists/playlists_service.cc

```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/playlists/playlists_service.h"
+
+#include <string>
+#include <utility>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/task/post_task.h"
+#include "base/task_runner_util.h"
+#include "brave/components/playlists/browser/playlists_controller.h"
+#include "content/public/browser/browser_context.h"
+
+#if !defined(OS_ANDROID)
+#include "brave/browser/playlists/desktop_playlists_player.h"
+#include "chrome/browser/profiles/profile.h"
+#endif
+
+namespace brave_playlists {
+namespace {
+
+const base::FilePath::StringType kBaseDirName(FILE_PATH_LITERAL("playlists"));
+
+}  // namespace
+
+PlaylistsService::PlaylistsService(content::BrowserContext* context)
+    : base_dir_(context->GetPath().Append(kBaseDirName)),
+#if !defined(OS_ANDROID)
+      context_(context),
+#endif
+      controller_(new PlaylistsController(context)),
+      weak_factory_(this) {}
+
+PlaylistsService::~PlaylistsService() {
+#if BUILDFLAG(ENABLE_EXTENSIONS)
+  if (playlists_event_router_)
+    controller_->RemoveObserver(playlists_event_router_.get());
+#endif
+
+  controller_->set_player(nullptr);
+}
+
+bool PlaylistsService::Init() {
+  return base::PostTaskAndReplyWithResult(
+      file_task_runner(), FROM_HERE,
+      base::BindOnce(&base::CreateDirectory, base_dir_),
+      base::BindOnce(&PlaylistsService::OnBaseDirectoryReady,
```

Comment:
creating the base directory before you write anything seems like it just unnecessarily complicates initialization. If we do write things to the filesystem (see https://github.com/brave/brave-core/pull/5733/files#r458326298) it seems like it would be better to call CreateBaseDirectoryIfNeeded before writing anything to avoid this